src/contrib/tests/unit-tests.rst

```diff
@@ -0,0 +1,25 @@
+.. _unit-tests:
+
+=============
+Unit tests
+=============
+
+Unit tests typically run a function or a tevent request without running the full deamon. The unit tests in SSSD are developed using either the `check <https://libcheck.github.io/check>`_ library or the `cmocka <https://cmocka.org>`_ library. Cmocka should be used for any new unit tests added to SSSD.
+
+The unit tests are fast to execute and in general this is where corner cases are typically easiest to test as you can provide false or unexpected input to the code under test. Unit tests are also often used to test a library's API.
+
+An important part of many tests using cmocka is wrapping a function provided by an external library using the `ld` linker's `--wrap` feature. You can learn more about cmocka and this feature in a `lwn.net article the cmocka developers contributed <https://lwn.net/Articles/558106/>`_. In the SSSD source tree, the unit tests reside under `src/tests/*.c` (check-based tests) and `src/tests/cmocka/*.c`.
+
```

Comment:
Add files and code into double backquotes:
```
``code``
``` 

This will make it an inline code.